compiler/packages/babel-plugin-react-compiler/src/HIR/Environment.ts

```diff
@@ -233,6 +233,51 @@ const EnvironmentConfigSchema = z.object({
 
   enableFunctionDependencyRewrite: z.boolean().default(true),
 
+  /**
+   * Enables inference of optional dependency chains. Without this flag
+   * a property chain such as `props?.items?.foo` will infer as a dep on
+   * just `props`. With this flag enabled, we'll infer that full path as
+   * the dependency.
+   */
+  enableOptionalDependencies: z.boolean().default(true),
```

Comment:
this is specific to auto-effect deps right? just curious since this is new, why not start with inferring optional deps by default?

Reply:
I think this was a bad merge